ssl/statem/statem_clnt.c

```diff
@@ -3336,7 +3336,12 @@ int tls_construct_client_certificate(SSL *s, WPACKET *pkt)
                     SSL3_CC_HANDSHAKE | SSL3_CHANGE_CIPHER_CLIENT_WRITE))) {
         SSLerr(SSL_F_TLS_CONSTRUCT_CLIENT_CERTIFICATE,
                SSL_R_CANNOT_CHANGE_CIPHER);
-        goto err;
+        /*
+         * This is a fatal error, which leaves
+         * enc_write_ctx in an inconsistent state
+         * and thus ssl3_send_alert may crash.
+         */
+        return 0;
```

Comment:
Looking at it again, I think that only a change_cipher_state with CC_WRITE can leave the
enc_write_ctx in an inconsistent state, and should be treated as a fatal error.
I assume that failure of change_cipher_state with CC_READ does not cause problems.

I only found one other place where a ssl3_send_alert is attempted after a failure of
change_cipher_state with CC_WRITE.

@mattcaswell was the code path where you saw the sending of an alert after a failed 
attempt to change the read cipher state?